core/src/commonMain/kotlin/io/github/aag/core/domain/repositories/LecturesRepository.kt

```diff
@@ -0,0 +1,12 @@
+package io.github.aag.core.domain.repositories
+
+import io.github.aag.core.domain.OperationResult
+import io.github.aag.core.domain.models.Lesson
+
+interface LecturesRepository {
+    suspend fun getAllLectures(courseId: Long): OperationResult<List<Lesson>, String?>
+
+    suspend fun getLecture(id: Long): Lesson
```

Comment:
Why some methods using `OperationResult<>` as a return type and others don't?

Reply:
Thanks for noticing! I think all should have `OperationResult` as a return type